### ISSUE TITLE
unitttest tests found with Python 3.2 but none found with 3.6

### ISSUE THREAD

**Issue by mloskot** [DonJayamanne/pythonVSCode#956]
## Environment data
VS Code version:  1.12.1
Python Extension version:  0.6.4
Python Version:   3.2 and 3.6
OS and version:    Windows 10 Pro

## Actual behavior

* Ediging `test_CreatePoint.py` file.
* Run *Python: Run Current Unit Test File*

* Using Python **3.2**, tests are discovered and run

![vscode-unittest-py32](https://cloud.githubusercontent.com/assets/80741/25990748/283ff7ca-3701-11e7-8971-a35cc4fd5a1a.png)

* Using Python **3.6**, no tests are found

![vscode-unittest-py36](https://cloud.githubusercontent.com/assets/80741/25990769/372be424-3701-11e7-9a38-caf2ceaa5aa5.png)

## Expected  behavior

* Python **3.6** should behave no different to Python **3.2**

## Settings  

`settings.json` is presented on both screenshots

## Logs

Output from ```Console window``` (Help->Developer Tools menu)

* Python **3.2** run

```
extensionHost.ts:285 [Extension Host] Error: This socket has been ended by the other party
    at Socket.writeAfterFIN [as write] (net.js:294:12)
    at sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:419:20)
    at JediProxy.sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:128:16)
    at JediProxyHandler.sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:548:24)
    at PythonHoverProvider.<anonymous> (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:104:54)
    at Generator.next (<anonymous>)
    at C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:7:71
    at __awaiter (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:3:12)
    at PythonHoverProvider.provideHover (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:82:16)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:306824
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:73477
    at new n.Class.derive._oncancel (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:69526)
    at Object.f [as asWinJsPromise] (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:73440)
    at e.provideHover (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:306778)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:316959
    at t._withAdapter (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:314586)
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] ---------------------------
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] sendCommand jediProxy
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] Error (sendCommand) This socket has been ended by the other party
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] Error: This socket has been ended by the other party
    at Socket.writeAfterFIN [as write] (net.js:294:12)
    at sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:419:20)
    at JediProxy.sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:128:16)
    at JediProxyHandler.sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:548:24)
    at PythonHoverProvider.<anonymous> (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:104:54)
    at Generator.next (<anonymous>)
    at C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:7:71
    at __awaiter (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:3:12)
    at PythonHoverProvider.provideHover (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:82:16)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:306824
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:73477
    at new n.Class.derive._oncancel (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:69526)
    at Object.f [as asWinJsPromise] (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:73440)
    at e.provideHover (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:306778)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:316959
    at t._withAdapter (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:314586)
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] Error: This socket has been ended by the other party
    at Socket.writeAfterFIN [as write] (net.js:294:12)
    at sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:419:20)
    at JediProxy.sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:128:16)
    at JediProxyHandler.sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:548:24)
    at PythonHoverProvider.<anonymous> (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:104:54)
    at Generator.next (<anonymous>)
    at C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:7:71
    at __awaiter (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:3:12)
    at PythonHoverProvider.provideHover (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:82:16)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:306824
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:73477
    at new n.Class.derive._oncancel (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:69526)
    at Object.f [as asWinJsPromise] (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:73440)
    at e.provideHover (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:306778)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:316959
    at t._withAdapter (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:314586)
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] ---------------------------
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] sendCommand jediProxy
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] Error (sendCommand) This socket has been ended by the other party
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] Error: This socket has been ended by the other party
    at Socket.writeAfterFIN [as write] (net.js:294:12)
    at sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:419:20)
    at JediProxy.sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:128:16)
    at JediProxyHandler.sendCommand (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\jediProxy.js:548:24)
    at PythonHoverProvider.<anonymous> (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:104:54)
    at Generator.next (<anonymous>)
    at C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:7:71
    at __awaiter (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:3:12)
    at PythonHoverProvider.provideHover (C:\Users\mateuszl\.vscode\extensions\donjayamanne.python-0.6.4\out\client\providers\hoverProvider.js:82:16)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:306824
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:73477
    at new n.Class.derive._oncancel (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:69526)
    at Object.f [as asWinJsPromise] (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:73440)
    at e.provideHover (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:306778)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:316959
    at t._withAdapter (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:314586)
d.logExtensionHostMessage @ extensionHost.ts:285
(anonymous) @ extensionHost.ts:139
emitTwo @ events.js:106
emit @ events.js:191
process.nextTick @ internal/child_process.js:752
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
extensionHost.ts:285 [Extension Host] Error: read ECONNRESET 
```

* Python **3.6** run  (Yes, it is that brief)

```
extensionHost.ts:285 [Extension Host] Error: read ECONNRESET 
```

**Comment by DonJayamanne** [comment]
@mloskot Apologies for never getting back to you on this.
I'd like to help you out with this issue. Currently I'm using 3.6 and haven't encountered such an issue (tested with 3.5, 3.4 and 2.7). 
Are you able to replicate this on any test file?

**Comment by mloskot** [comment]
@DonJayamanne No worries. Thanks for coming back.

I just checked with 3.2 in VSCode 1.16.1 with the latest Python ext. 0.7.0, and the issue seem a bit random.

1. Python: Run Current Unit Test File
   * The only output in Python Test Log is this
   * Developer Tools console does not print any errors

```
start
utility.test_Sample_Unit.TestSample.test_all
```

2. Python: Run All Unit Tests
  * Despite there are two test .py scripts, only currently open file tests run
  * Developer Tools console prints lots of errors (screenshot)

```
test_all (utility.test_Sample_Unit.TestSample) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.000s

OK
```

![vscode-python-unittest](https://user-images.githubusercontent.com/80741/31164918-4e89d900-a8e9-11e7-9f97-2ec90764d537.png)


Here is sample project I used to run the above

[vscode-py32-unittest.zip](https://github.com/DonJayamanne/pythonVSCode/files/1354947/vscode-py32-unittest.zip)
